tests/test_modutils.py

```diff
@@ -175,6 +175,41 @@ def test_import_symlink_with_source_outside_of_path(self) -> None:
             finally:
                 os.remove(linked_file_name)
 
+    def test_modpath_from_file_path_order(self) -> None:
+        """Test for ordering of paths.
+        The test does the following:
+        1. Add a tmp directory to beginning of sys.path
+        2. Create a module file in sub directory of tmp directory
+        3. If the sub directory is passed as additional directory, module name
+           should be relative to the subdirectory since additional directory has
+           higher precedence."""
+        orig_path = sys.path.copy()
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            try:
+                mod_name = "module"
+                sub_dirname = "subdir"
+                sub_dir = tmp_dir + "/" + sub_dirname
+                os.mkdir(sub_dir)
+                module_file = f"{sub_dir}/{mod_name}.py"
+
+                sys.path.insert(0, str(tmp_dir))
+                with open(module_file, "w+", encoding="utf-8"):
+                    pass
+
+                # Without additional directory, return relative to tmp_dir
+                self.assertEqual(
+                    modutils.modpath_from_file(module_file), [sub_dirname, mod_name]
+                )
+
+                # With sub directory as additional directory, return relative to
+                # sub directory
+                self.assertEqual(
+                    modutils.modpath_from_file(f"{sub_dir}/{mod_name}.py", [sub_dir]),
+                    [mod_name],
+                )
+            finally:
```

Comment:
I think elsewhere we use a context manager for sys path modification in tests. Sorry I can't search as I'm on mobile.

Reply:
Will check. Don't remember seeing one in `astroid` repo. There is one in `pylint` repo. Perhaps you are remembering that one? If there isn't one, you want me to create one similar to that?

Reply:
Didn't find it. Will create one.

Reply:
Probably this:

https://github.com/pylint-dev/pylint/blob/d9b06c6d5ba0fd9b74b5d8441accedc6bdbecfd8/pylint/lint/utils.py#L128-L135

Reply:
Sorry I did not realize it was an astroid PR 😅

Reply:
Updated